File: ts/src/grow_up_number.ts
const grow_up_number = (number: number) => {
  const result: string[] = [];
  for (let i = number; i >= 0; i--) result.push(i.toString());
	if (number === 0) return "[]";
  return result;
};

console.log(grow_up_number(5)); // ["5", "4", "3", "2", "1", "0"]
console.log(grow_up_number(3)); // ["3", "2", "1", "0"]
console.log(grow_up_number(0)); // []

File: rust/src/grow_up_number.rs
use std::vec::Vec;
use std::string::String;

pub fn grow_up_numbers(number: i32) -> Vec<String> {
  if number == 0 {
    return vec!["".to_string()];
  }

  (0 ..= number).rev().map(|i| i.to_string()).collect()
}